server/swap/swap.go

```diff
@@ -1049,123 +1051,46 @@ func (s *Swapper) tryConfirmSwap(status *swapStatus, confTime time.Time) {
 // not evaluated here, but in (Swapper).checkInaction. This method simply sets
 // the appropriate flags in the swapStatus structures.
 func (s *Swapper) processBlock(block *blockNotification) {
-	var completions []*matchTracker
-	s.matchMtx.Lock()
-	defer s.matchMtx.Unlock()
-	for _, match := range s.matches {
+	checkMatch := func(match *matchTracker) {
 		// If it's neither of the match assets, nothing to do.
-		if match.makerStatus.swapAsset != block.assetID && match.takerStatus.swapAsset != block.assetID {
-			continue
+		if match.makerStatus.swapAsset != block.assetID &&
+			match.takerStatus.swapAsset != block.assetID {
+			return
 		}
 
 		// Lock the matchTracker so the following checks and updates are atomic
 		// with respect to Status.
 		match.mtx.RLock()
+		defer match.mtx.RUnlock()
 
-	statusSwitch:
 		switch match.Status {
-		case order.NewlyMatched:
 		case order.MakerSwapCast:
 			if match.makerStatus.swapAsset != block.assetID {
-				break statusSwitch
+				break
 			}
 			// If the maker has broadcast their transaction, the taker's broadcast
 			// timeout starts once the maker's swap has SwapConf confs.
-			s.tryConfirmSwap(match.makerStatus, block.time)
+			if s.tryConfirmSwap(match.makerStatus, block.time) {
+				s.unlockOrderCoins(match.Maker)
```

Comment:
Now I'm rethinking this because of the possibility for never calling processBlock in one of these statuses.  Perhaps best just to do this on match complete like before.